src/qibolab/instruments/oscillator.py

```diff
@@ -1,4 +1,13 @@
-from qibolab.instruments.abstract import Instrument
+from dataclasses import dataclass
+from typing import Optional
+
+from qibolab.instruments.abstract import Instrument, InstrumentSettings
+
+
+@dataclass
+class LocalOscillatorSettings(InstrumentSettings):
+    power: Optional[float] = None
+    frequency: Optional[float] = None
```

Comment:
@stavros11 are them really optional?

Reply:
> @stavros11 are them really optional?

They are needed for execution. However, we initialize them as `None` so that we can instantiate the instrument without passing them in the platform `create` methods and assign their values later using `instrument.setup` (or the setters) after loading the runcard.

In principle we could initialize the instrument with `self.settings = None` and then instantiate `settings` when `setup` or the setters are called, but it is almost the same and maybe slightly more complicated in terms of implementation.

Reply:
On the one side, I'd agree with you for the current case. But if we had many more settings, the current one is scaling bad (you need linearly more code), while your second solution does not scale at all :)
(i.e. the second solution has arguably no overhead, or linearly less than the other one)